core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/Debian/setup.yml

```diff
@@ -20,13 +20,15 @@
     - not custom_repository_url
     - inventory_hostname in target_repository_hostnames
 
-- name: Include Helm binary installation 
-  include_role:
-    name: helm
-    tasks_from: install-bin
-
-- name: Include helm repository creation
-  include_tasks: "create-helm-repo.yml"
+- name: Helm binary installation and Helm repository creation
+  block:
+    - name: Include Helm binary installation 
+      include_role:
+        name: helm
+        tasks_from: install-bin
+    - name: Include Helm repository creation
+      include_tasks: "create-helm-repo.yml"
+  when: inventory_hostname in groups['kubernetes_master'][0]
```

Comment:
Please consider using the `target_repository_hostnames[0]` defined here https://github.com/epiphany-platform/epiphany/blob/cc57b54f4e3694aa8a6b55217bf8a72a9191d726/core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/gather-facts.yml
The cause for this is currently we have repository on single host (for performance reasons) and this host can theoretically switch places inside the kubernetes_master group in inventory (for example in AWS while adding more master nodes to kubernetes cluster) with other nodes.